### ISSUE TITLE
Accessibility Improvement: Add labels to theme switcher

### ISSUE THREAD

**Issue by 0x3mr** [OneBusAway/wayfinder#206]
The theme switcher component lacks proper accessibility as in 'labels' which makes it difficult for users with assistive technologies to access the website easily such as a screen reader.

- The `<input type="checkbox">` in `ThemeSwitcher.svelte` does not have an associated accessible label.
- Screen readers may not correctly announce its purpose.

## Current accessibility audit report:
![Image](https://github.com/user-attachments/assets/a44c802d-7c87-4d12-8e54-b7555523c752)

## Suggested fix:

Add `aria-label="Toggle theme` to the checkbox input to improve screen reader support.

## Expected results:
Increase by 5% in accessibility audit report.
![Image](https://github.com/user-attachments/assets/b67c9208-455b-4f2b-ae68-aa0c43d52377)
